topcoder-platform/freeCodeCamp
# PROD-2031 Passthrough TC/FCC Auth -> PROD-1985_topcoder-academy-prototype

https://topcoder.atlassian.net/browse/PROD-2211

This PR updates the FCC app so that it uses the TC Auth0 client. This means that if you are logged into the Learn app then navigate to a lesson, FCC will recognize the user and will track progress against that user.

This PR only addresses the scenario that the TC user already has a linked FCC account. PROD-2289 will address adding new FCC users.

This source branch is off main, so it doesn't include any of the UI changes for TC. This was done to simplify troubleshooting as well as to be able to see/use the Sign In button.

**To verify this PR:**

1. set up your .env file to match sample.topcoder.env
2. set the JWT_SECRET in the .env as the Auth0 signing certificate (using \n for line breaks--can get from brooke)
3. log into the customer work app as a user that already has an account in the FCC (if you need to create an existing user, see instructions below)
4. navigate to a lesson
5. if it works correctly, you should not see the Sign in link in the top nav
6. complete a new lesson
7. you should see this completed lesson in the DB
8. you should also see this completed lesson when you go to the FCC as a standalone app

**To create a new FCC user:**

1. log into the Learn app
2. copy the tcjwt cookie value
3. decode the [jwt](https://jwt.io/)
4. note the "sub" property (e.g. auth0|40029484)
5. log out of the Learn app
6. check out the **main** branch of the FCC
7. change .env vars to match sample.topcoder.env
8. change the following .env vars 

AUTH0_CLIENT_SECRET=/Auth0 CLIENT SECRET/
JWT_SECRET=not-applicable

9. run the FCC as a standalone app
10. Click Sign in
11. Submit credentials for an existing TC user
12. Return to the FCC home page
13. you should not see the Sign in link
14. Find the new user in the Mongo DB
15. change the value of the externalId property to match the value you noted in step 4
16. check out the PROD-passthrough branch
17. reload the FCC as a standalone
18. you should still not the Sign in button
19. complete a new lesson
20. you should see the user updated in the db
